### PR TITLE
Feat/348166  manage court

### PR DESCRIPTION
**Pull request by whitewaterdesign**
API changes for add / remove a court

### CODE REVIEW

File: app/repos/courts.js [hunk 1]
```diff
@@ -13,6 +17,86 @@ const getCourts = async () => {
...
+ * @param [transaction]
+ * @returns {Promise<*|undefined>}
+ */
+const createCourt = async (courtData, user, transaction) => {
+  if (!transaction) {
+    return await sequelize.transaction(async (t) => createCourt(courtData, user, t))
+  }
+  const foundCourt = await sequelize.models.court.findOne({
+    where: {
+      name: courtData.name
```
Comment (jbarnsley10) [inline app/repos/courts.js:36]: Does this automatically do a case-insensitive search? It does with SQL Server but think Postgres is case-sensitive

File: app/repos/courts.js [hunk 2]
```diff
@@ -13,6 +17,86 @@ const getCourts = async () => {
...
+    }
+  })
+
+  if (foundCourt !== null) {
+    throw new DuplicateResourceError(`Court with name ${courtData.name} already exists`)
+  }
+
+  let court
+
+  const foundParanoid = await sequelize.models.court.findOne({
```
Comment (jbarnsley10) [inline app/repos/courts.js:46]: This is good

File: app/repos/courts.js (hunk 1 above)
Comment (jbarnsley10) [reply app/repos/courts.js:36]: If you use the 'like' and %% then it is case insensitive (see lookups methods)

File: app/routes/courts.js [hunk 3]
```diff
@@ -1,13 +1,49 @@
...
-  method: 'GET',
-  path: '/courts',
-  handler: async (request, h) => {
-    const courts = await getCourts()
+module.exports = [
+  {
+    method: 'GET',
+    path: '/courts',
+    handler: async (request, h) => {
+      const courts = await getCourts()
```
Comment (jbarnsley10) [inline app/routes/courts.js:10]: Do you not need to query the courts to include soft-deleted records? Can't remember if possible to re-enable a court

**Review by jbarnsley10 — CHANGES_REQUESTED** [review]
Looks great. A couple of questions though

File: app/routes/courts.js (hunk 3 above)
Comment (whitewaterdesign) [reply app/routes/courts.js:10]: I don't think so, we aiming to not show them aren't we on the dropdowns?

File: app/repos/courts.js (hunk 1 above)
Comment (whitewaterdesign) [reply app/repos/courts.js:36]: very good point

File: app/routes/courts.js (hunk 3 above)
Comment (whitewaterdesign) [reply app/routes/courts.js:10]: The api will now handle create of old records.  It's possible a query param could be added in future to this to get soft deleted records as well, but the consensus was no list 🤷.  Reckon we can update when needed.  This ticket has already been downgraded, so trying not worth the extra work at this stage IMO.

**Review by jbarnsley10 — APPROVED** [review]
LGTM

**Pull request merged**